packages/attribute-registry-process/src/services/readModelService.ts

```diff
@@ -1,5 +1,223 @@
-/* import { logger, ReadModelRepository } from "pagopa-interop-commons";
+import { AggregationCursor } from "mongodb";
+import { z } from "zod";
+import { logger, ReadModelRepository } from "pagopa-interop-commons";
+import {
+  AttributeKind,
+  AttributeTmp,
+  Document,
+  ErrorTypes,
+  WithMetadata,
+} from "pagopa-interop-models";
+import { ListResult } from "../model/types.js";
 import { config } from "../utilities/config.js";
-const { attributes } = ReadModelRepository.init(config); */
 
-// TODO
+const { attributes } = ReadModelRepository.init(config);
+
+function arrayToFilter<T, F extends object>(
+  array: T[],
+  f: (array: T[]) => F
+): F | undefined {
+  return array.length > 0 ? f(array) : undefined;
+}
+
+async function getTotalCount(
+  query: AggregationCursor<Document>
+): Promise<number> {
+  const data = await query.toArray();
+  const result = z.array(z.object({ count: z.number() })).safeParse(data);
+
+  if (result.success) {
+    return result.data.length > 0 ? result.data[0].count : 0;
+  }
+
+  logger.error(
+    `Unable to get total count from aggregation pipeline: result ${JSON.stringify(
+      result
+    )} - data ${JSON.stringify(data)} `
+  );
+  throw ErrorTypes.GenericError;
+}
+
+export const readModelService = {
+  async getAttributes(
+    {
+      kinds,
+      name,
+      origin,
+    }: {
+      kinds: AttributeKind[];
+      name?: string | undefined;
+      origin?: string | undefined;
```

Comment:
is this equivalent to
```suggestion
      name?: string;
      origin?: string;
```
?